torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -69,13 +68,28 @@ ncclDataType_t getNcclDataType(at::ScalarType type) {
 }
 
 ncclRedOp_t getNcclReduceOp(const ReduceOp reduceOp, at::Tensor& input) {
-  if (reduceOp == ReduceOp::SUM && input.scalar_type() == at::kBool) {
-    // For bool tensors, map sum to max, which both represent a bitwise or.
-    // This is to prevent overflow issues with sum, since we use uint8 to
-    // represent a bool (see ncclDataType mapping).
-    return ncclMax;
+  switch (reduceOp) {
+    case ReduceOp::BAND:
+      throw std::runtime_error("Cannot use ReduceOp.BAND with NCCL");
+      break;
+    case ReduceOp::BOR:
+      throw std::runtime_error("Cannot use ReduceOp.BOR with NCCL");
+      break;
+    case ReduceOp::BXOR:
+      throw std::runtime_error("Cannot use ReduceOp.BXOR with NCCL");
+      break;
+    case ReduceOp::UNUSED:
+      throw std::runtime_error("Unhandled ReduceOp");
+      break;
+    default:
+      if (reduceOp == ReduceOp::SUM && input.scalar_type() == at::kBool) {
+        // For bool tensors, map sum to max, which both represent a bitwise or.
+        // This is to prevent overflow issues with sum, since we use uint8 to
+        // represent a bool (see ncclDataType mapping).
+        return ncclMax;
+      }
+      return ncclOp[reduceOp];
```

Comment:
Could you also mark the map as const and use the `find()` accessor to return this as suggested in https://github.com/pytorch/pytorch/issues/39708?